cip/1.accepted/CIP2015-10-27-State-visibility-between-clauses.adoc

```diff
@@ -16,27 +16,35 @@ Currently, because this is not explicit enough, the behaviour is implementation
 toc::[]
 
 === Background
-Cypher allows queries to go through multiple steps, and it’s possible to interleave reading and writing to the graph.
+
+Cypher allows queries to go through multiple steps, and it's possible to interleave reading and writing to the graph.
 Some clauses can also do both - read the graph and write to it at the same time.
 
 To allow the user to understand what the outcome will be at an abstract, logical level, semantics for the visibility of these changes should be clear.
 
 Note:
 This is not about concurrency - even in a single user system this would need to be thought through and defined explicitly.
 
+===== Definition
+
+We call the concept discussed in this CIP _Eagerness_.
+The meaning here is that each clause is logically consuming _all_ incoming rows _eagerly_, without producing any output until the input is exhausted.
+This concept is the opposite of link:https://en.wikipedia.org/wiki/Lazy_evaluation[_lazy evaluation_], which applied to Cypher means that each clause only processes as much input as is necessary for the first record of output to be sent to the subsequent clause.
```

Comment:
"This concept is the opposite of" -> "This concept is in contrast to"

"which applied to Cypher means that each clause only processes as much input as is necessary for the first record of output to be sent to the subsequent clause."
->
"which, when applied to Cypher, means that each clause only processes as much input as is necessary for the first record of output to be sent to the subsequent clause."